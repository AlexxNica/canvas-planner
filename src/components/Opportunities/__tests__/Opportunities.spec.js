import React from 'react';
import { shallow } from 'enzyme';
import { Opportunities } from '../index';

function defaultProps (option) {
  return {
    opportunities: [{id: 1, course_id: 1, due_at: "2017-03-09T20:40:35Z", html_url: "http://www.non_default_url.com", name: "learning object title"}],
    courses: [{id: 1, shortName: "Course Short Name"}],
    timeZone: 'America/Denver',
  };
}

it('renders the base component correctly with one opportunity', () => {
  const wrapper = shallow(
    <Opportunities {...defaultProps()} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders the right course with the right opportunity', () => {
  let tempProps = defaultProps();
  tempProps.opportunities = tempProps.opportunities.concat({id: 2, course_id: 2, html_url: "http://www.non_default_url.com", due_at: "2017-03-09T20:40:35Z", name: "other learning object"});
  tempProps.courses = tempProps.courses.concat({id: 2, shortName: "A different Course Name"});
  const wrapper = shallow(
    <Opportunities {...tempProps} />
  );
  expect(wrapper).toMatchSnapshot();
});

