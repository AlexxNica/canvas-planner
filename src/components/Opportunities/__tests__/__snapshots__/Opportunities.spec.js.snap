// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing if no opportunities 1`] = `
<ol>
  Education is what remains after one has forgotten what one has learned in school.
</ol>
`;

exports[`renders the base component correctly with one opportunity 1`] = `
<ol>
  <li>
    <Opportunity
      courseName="Course Short Name"
      dismiss={[Function]}
      dueAt="2017-03-09T20:40:35Z"
      id="1"
      opportunityTitle="learning object title"
      timeZone="America/Denver"
      url="http://www.non_default_url.com"
    />
  </li>
</ol>
`;

exports[`renders the right course with the right opportunity 1`] = `
<ol>
  <li>
    <Opportunity
      courseName="Course Short Name"
      dismiss={[Function]}
      dueAt="2017-03-09T20:40:35Z"
      id="1"
      opportunityTitle="learning object title"
      timeZone="America/Denver"
      url="http://www.non_default_url.com"
    />
  </li>
  <li>
    <Opportunity
      courseName="A different Course Name"
      dismiss={[Function]}
      dueAt="2017-03-09T20:40:35Z"
      id="2"
      opportunityTitle="other learning object"
      timeZone="America/Denver"
      url="http://www.non_default_url.com"
    />
  </li>
</ol>
`;
